snap/snapcraft.yaml

```diff
@@ -0,0 +1,109 @@
+---
+name: open-webui
+base: core22
+version: '0'  # no released version yet
+summary: ChatGPT-Style Web UI Client for Ollama 🦙 (formerly Ollama WebUI)
+description: |
+  The Open WebUI consists of two primary components:
+  the frontend and the backend (which serves as a reverse proxy,
+  handling static frontend files, and additional features).
+  Both are set up.
+# adopt-info: backend
+grade: stable
+confinement: strict
+license: MIT
+architectures:
+  - build-on: [amd64]
+    build-for: [amd64]
+  - build-on: [arm64]
+    build-for: [arm64]
+
+apps:
+  open-webui:
+    command: bin/run-snap.sh
+    plugs:
+      - home
+      - removable-media
+      - network
+      - network-bind
+      - opengl
+    environment:
+      ENV: prod
+      SCARF_NO_ANALYTICS: 'true'
+      DO_NOT_TRACK: 'true'
+  listener:
+    command: bin/run-snap.sh
+    daemon: simple
+    install-mode: enable
+    restart-condition: on-failure
+    plugs:
+      - home
+      - removable-media
+      - network
+      - network-bind
+      - opengl
+    environment:
+      ENV: prod
+      SCARF_NO_ANALYTICS: 'true'
+      DO_NOT_TRACK: 'true'
+parts:
+  launcher:
+    plugin: dump
+    source: ./snap/local
+    organize:
+      run-snap.sh: bin/run-snap.sh
+  minilm:
+    plugin: dump
+    source: https://chroma-onnx-models.s3.amazonaws.com/all-MiniLM-L6-v2/onnx.tar.gz
+    organize:
+      onnx: root/.cache/chroma/onnx_models/all-MiniLM-L6-v2/onnx
+  frontend:
+    after: [launcher, minilm]
+    plugin: nil
+    source: .
+    source-type: git
+    override-build: |
+      craftctl default
+      npm ci
+      npm run build
+      cp -r build $CRAFT_PART_INSTALL/frontend
+    stage-snaps:
+      - node/20/stable
+  backend:
+    after: [frontend]
+    plugin: python
+    source: .
+    source-type: git
+    build-packages:
+      - netcat-openbsd
+      - python3.10
```

Comment:
Correct me if I'm wrong, but I believe we should update this part to match our docker container image python environment (python3.11).

Reply:
Oh yes, that makes sense.

Reply:
I've pushed an update to the beta channel built with python 3.11.